Source/santad/SNTApplicationTest.m

```diff
@@ -0,0 +1,129 @@
+/// Copyright 2021 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+#import <EndpointSecurity/EndpointSecurity.h>
+#import <MOLCertificate/MOLCertificate.h>
+#import <MOLCodesignChecker/MOLCodesignChecker.h>
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
+
+#import "Source/common/SNTConfigurator.h"
+#import "Source/santad/SNTApplication.h"
+#import "Source/santad/SNTDatabaseController.h"
+
+#include "Source/santad/EventProviders/EndpointSecurityTestUtil.h"
+
+@interface SNTApplicationTest : XCTestCase
+@property id mockSNTDatabaseController;
+@end
+
+@implementation SNTApplicationTest
+- (void)setUp {
+  [super setUp];
+  fclose(stdout);
+  self.mockSNTDatabaseController = OCMClassMock([SNTDatabaseController class]);
+  XCTAssertTrue([[SNTConfigurator configurator] enableSystemExtension]);
+}
+
+- (void)tearDown {
+  [self.mockSNTDatabaseController stopMocking];
+  [super tearDown];
+}
+
+- (void)checkBinaryExecution:(NSString *)binaryName
+                    testPath:(NSString *)testPath
+                  wantResult:(es_auth_result_t)wantResult {
+  MockEndpointSecurity *mockES = [MockEndpointSecurity mockEndpointSecurity];
+  [mockES reset];
+
+  OCMStub([self.mockSNTDatabaseController databasePath]).andReturn(testPath);
+
+  SNTApplication *app = [[SNTApplication alloc] init];
+  [app start];
+
+  // es events will start flowing in as soon as es_subscribe is called, regardless
+  // of whether we're ready or not for it.
+  while (!mockES.subscribed)
+    ;
```

Comment:
Limit this to a small number of seconds to avoid long hangs if something isn't working.

```objc
dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
dispatch_async(dispatch_get_global_queue(QOS_CLASS_BACKGROUND, 0), ^{
  while (!mockES.subscribed);
  dispatch_semaphore_signal(semaphore);   
});
XCTAssertEqual(dispatch_semaphore_wait(semaphore, 5 * NSEC_PER_SEC), 0, @"failed to subscribe");
```

Reply:
Done but with an XCTest Expectation instead of a semaphore.